### DISCUSSION TITLE
"single-class" detector: objectness loss?

### DISCUSSION THREAD

**Discussion by simonespring** [ultralytics/yolov5#12872] [Q&A]
Hi all, I have a question regarding the training of a single-class yolov5 object detection model.

Since the model should learn how to only detect a single specific class, the class loss is identically zero throughout the training process.  But what about the objectness loss? Should it be included in the computation of the "total" loss function although I have a single class only?

Also, note that during the training process, the evolution of the objectness score is quite peculiar. In particular, while the box loss decreases "as expected", this is not true for the objectness loss. In particular, the objectness loss computed on the training dataset decreases almost linearly, while the objectness loss computed on the validation dataset reaches a sort of "plateau" (see attached plot).

Do you have any recommendation/suggestion about how to improve the training performances?

Any advice would be great, thank you in advance!

![training](https://github.com/ultralytics/yolov5/assets/119604043/3b14d87f-5290-4210-9a8f-a82e59c55601)
